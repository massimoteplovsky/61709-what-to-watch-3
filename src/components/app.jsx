import React from "react";
import Main from "./main.jsx";

// eslint-disable-next-line react/prop-types
const App = ({filmData}) => {
  return (
    <Main filmData={filmData}/>
  );
};

export default App;
