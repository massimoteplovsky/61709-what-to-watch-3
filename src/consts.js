export const DEFAULT_GENRE = `All genres`;
export const FILM_TO_SHOW = 8;
export const RELATED_FILM_COUNT = 4;
export const AUTH = `AUTH`;
export const NO_AUTH = `NO_AUTH`;
export const BASE_URL = `https://htmlacademy-react-2.appspot.com`;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const NOT_FAVORITE_FILM = 0;
export const FAVORITE_FILM = 1;
export const MAX_RATING = 5;
