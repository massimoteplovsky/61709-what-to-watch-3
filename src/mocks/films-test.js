export const films = [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    genre: [`Action`],
    year: 1991,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 2,
    title: `1+1`,
    genre: [`Drama`, `Comedy`],
    year: 2012,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 3,
    title: `Joker`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 6,
    title: `1917`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 7,
    title: `Parasite`,
    genre: [`Drama`, `Comedy`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 8,
    title: `Irishman`,
    genre: [`drama`, `criminal`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  }
];

export const film = {
  id: 1,
  title: `Terminator 2. Judgment day`,
  genre: [`Action`],
  year: 1991,
  promoPoster: ``,
  poster: ``,
  cover: ``,
  description: ``,
  rating: 8.5,
  votes: 12000,
  director: ``,
  starring: []
};

export const promoFilmInfo = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};
