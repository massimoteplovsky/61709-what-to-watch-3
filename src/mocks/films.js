export default [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    poster: `http://i4.ytimg.com/vi/HgV7-MJwUBw/mqdefault.jpg`
  },
  {
    id: 2,
    title: `1+1`,
    poster: `https://i.pinimg.com/736x/76/44/c0/7644c07844dd4cb28d2c289bfa6e0dec--the-intouchables-comedy-movies.jpg`
  },
  {
    id: 3,
    title: `Joker`,
    poster: `https://all.culture.ru/uploads/9b743d135092e110a8e55a3dc3ad5bc4_w375_h250.jpeg`
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    poster: `https://avatars.mds.yandex.net/get-pdb/1945878/3281b62e-4539-4eb7-a26b-fa68f6572bea/s375`
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    poster: `https://sun9-28.userapi.com/c206816/v206816832/1771c/3p_wj8sfMz8.jpg`
  },
  {
    id: 6,
    title: `1917`,
    poster: `https://sun3-13.userapi.com/c857724/v857724955/14e882/u5CFMvbvX3I.jpg`
  },
  {
    id: 7,
    title: `Parasite`,
    poster: `http://m.kino-teatr.ru/art/5395/pv_75950.jpg`
  },
  {
    id: 8,
    title: `Irishman`,
    poster: `https://sun9-31.userapi.com/c850608/v850608676/40eb4/ojB5App6exk.jpg`
  },
];
