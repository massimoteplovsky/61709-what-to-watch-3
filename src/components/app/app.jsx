import React, {PureComponent} from "react";
import {Router, Switch, Route, Redirect} from "react-router-dom";
import history from "../../history.js";
import {PropTypes} from "prop-types";
import {connect} from "react-redux";
import Main from "../main/main.jsx";
import Movie from "../movie/movie.jsx";
import Signin from "../signin/signin.jsx";
import VideoPlayer from "../video-player/video-player.jsx";
import AddReview from "../add-review/add-review.jsx";
import MyList from "../my-list/my-list.jsx";
import ServerError from "../server-error/server-error.jsx";
import PageNotFound from "../page-not-found/page-not-found.jsx";
import PrivateRoute from "../private-route/private-route.jsx";
import withVideoPlayer from "../../hocs/with-video-player/with-video-player.js";
import withLoading from "../../hocs/with-loading/with-loading.js";
import {getAuthorizationStatus} from "../../selectors/user/user.js";
import {AUTH} from "../../consts.js";
import {loadAllFilms, loadPromoFilm} from "../../actions/action-creators/film-actions/film-actions.js";
import {checkAuth} from "../../actions/action-creators/user-actions/user-actions.js";

const WrappedVideoPlayer = withVideoPlayer(VideoPlayer);

class App extends PureComponent {

  componentDidMount() {
    const {
      onChangeLoadingStatus,
      onLoadData
    } = this.props;

    onLoadData().then(() => {
      onChangeLoadingStatus(false);
    });
  }

  render() {
    const {
      loading,
      authorizationStatus,
      error,
    } = this.props;

    if (error) {
      return <ServerError/>;
    }

    if (loading) {
      return null;
    }

    return (
      <Router history={history}>
        <Switch>
          <Route
            path="/"
            exact
            component={Main}
          />
          <Route
            path="/login"
            exact
            render={() => {
              return (
                authorizationStatus === AUTH ?
                  <Redirect to="/"/>
                  :
                  <Signin/>
              );
            }}
          />
          <Route
            path="/films/:id"
            exact
            render={(props) => {
              return <Movie {...props}/>;
            }}
          />
          <PrivateRoute
            exact
            path="/films/:id/review"
            render={(filmID) => {
              return <AddReview filmID={filmID}/>;
            }}
          />
          <Route
            path="/player/:id"
            exact
            render={(props) => {
              return (
                <WrappedVideoPlayer
                  isPlaying={true}
                  isMuted={false}
                  filmID={props.match.params.id}
                  isPreviewMode={false}
                />
              );
            }}
          />
          <PrivateRoute
            path="/myList"
            exact
            render={() => <MyList/>}
          />
          <Route component={PageNotFound}/>
        </Switch>
      </Router>
    );
  }
}

App.propTypes = {
  error: PropTypes.bool.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  onChangeLoadingStatus: PropTypes.func.isRequired,
  onLoadData: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  match: PropTypes.object
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  error: state.application.error
});

const mapDispatchToProps = (dispatch) => ({
  onLoadData() {
    return Promise.all([
      dispatch(loadAllFilms()),
      dispatch(loadPromoFilm()),
      dispatch(checkAuth())
    ]);
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(withLoading(App));
