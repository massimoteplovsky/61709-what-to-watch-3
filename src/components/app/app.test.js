import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app.jsx";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {films} from "../../mocks/films-test";
import {NO_AUTH} from "../../consts.js";

const mockStore = configureMockStore([]);
let store = mockStore({
  films: {
    filteredFilms: films,
    favoriteFilms: films,
    films,
    promoFilm: null,
    filmCounter: 8,
    actualGenre: ``,
    reviews: []
  },
  application: {
    error: false
  },
  user: {
    authorizationStatus: NO_AUTH,
    userInfo: null,
  }
});

it(`<App /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            error={false}
            authorizationStatus={NO_AUTH}
            onChangeLoadingStatus={() => {}}
            onLoadData={() => Promise.resolve()}
            loading={false}
            match={{}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`<App /> component with error status is true renders error`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            error={true}
            authorizationStatus={NO_AUTH}
            onChangeLoadingStatus={() => {}}
            onLoadData={() => Promise.resolve()}
            loading={false}
            match={{}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});


