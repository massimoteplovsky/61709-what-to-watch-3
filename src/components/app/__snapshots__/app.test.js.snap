// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> component renders correctly 1`] = `null`;

exports[`<App /> component with error status is true renders error 1`] = `
<div
  className="user-page"
>
  <h1>
    Sorry, connection with server failed!
  </h1>
</div>
`;
